packages/svelte/src/internal/client/runtime.js

```diff
@@ -963,7 +963,10 @@ export function set_signal_value(signal, value) {
 		// If we have afterUpdates locally on the component, but we're within a render effect
 		// then we will need to manually invoke the beforeUpdate/afterUpdate logic.
 		// TODO: should we put this being a is_runes check and only run it in non-runes mode?
-		if (current_effect === null && current_queued_pre_and_render_effects.length === 0) {
+		if (
+			current_effect === null &&
+			current_queued_pre_and_render_effects.every((e) => e.context !== component_context)
```

Comment:
Feel like this is even more readable.
Also I don't understand why this fixes something, a short comment might help future us
```suggestion
			current_queued_pre_and_render_effects.some((e) => e.context === component_context)
```

Reply:
They're not equivalent :) 1 and 2 mean the same thing, 3 means something different:

### 1

```js
current_queued_pre_and_render_effects.filter((e) => e.context === component_context).length === 0
```

'the number of effects whose context is `component_context` is zero'

### 2 

```js
current_queued_pre_and_render_effects.every((e) => e.context !== component_context)
```

'for every effect, it is true that the context is not `component_context`'

### 3

```js
current_queued_pre_and_render_effects.some((e) => e.context === component_context)
```

'there is one or more effect whose context is `component_context`'.

If I make that change the test fails. Which makes sense when you put it like this:

```js
numbers = [1, 2, 3];
numbers.filter(n => n === 4).length === 0; // true
numbers.every(n => n !== 4); // true
numbers.some(n => n === 4); // false
```

Agree that this could use a comment though. I have no idea why this works

Reply:
(also, `[].every(predicate)` and `[].some(predicate)` will always yield `true` and `false` respectively, regardless of `predicate`)

Reply:
We're checking for any render effects that are "local", i.e. have the same context as the source signal. If we have them, then the logic in the render effect phase will handle this logic for us for when they update. However, if we only have render effects from another component, then there won't be a render effect that handles the `beforeUpdate` and `afterUpdate` for this component context – so we need to manually invoke it.

Reply:
I'll switch to 2) and add a comment.